lib/Analyze.cpp

```diff
@@ -920,8 +979,9 @@ void RecoverMemoryAccesses(
 
 void ReplaceImmediateIntegers(
     const Program &program,
-    const std::vector<std::pair<llvm::Use *, uint64_t>> &ci_fixups) {
-  for (auto [use, imm_val] : ci_fixups) {
+    const std::vector<std::tuple<llvm::Use *, uint64_t, llvm::Type *>>
+        &ci_fixups) {
+  for (auto [use, imm_val, type] : ci_fixups) {
```

Comment:
I think we missed an opportunity here. I think below, just before `if (new_val != used_val)` we should check `if (type && type->isPointerTy())` then do the `call` injection here as `new_val`.